### ISSUE TITLE
Getting java.io.IOException: Cannot create file `file://<tablePath>` when using Flink/Delta Connector with S3

### ISSUE THREAD

**Issue by pabloFloresHdz** [delta-io/connectors#340]
Hi! I'm getting  a `java.io.IOException: Cannot create file file:/<tablePath>` when the delta connector for flink tries to write the DeltaLog for a DataLake hosted in an S3 bucket. I'm using the delta-flink v0.4.0 connector.

Two things came to my attention: 

1. The connector is trying to write the delta log in a local directory instead of the S3 bucket
2. The delta table path is different than the one I specified. I'm using a table path like `s3a://<bucket>/<some>/<path>` and the path that the DeltaLog is trying to access is something like `file:/<some>/<path>`

I'm using something similar to the following code to initialize the DeltaSink

```scala
private val schema = new RowType(
    Seq(new RowType.RowField("text", new VarCharType(VarCharType.MAX_LENGTH)),
      new RowType.RowField("date", new VarCharType(VarCharType.MAX_LENGTH)),
    ).asJava
  )
def createDeltaSink: DeltaSink[RowData] = {
    val hadoopConf = new Configuration()
    hadoopConf.set("io.delta.standalone.LOG_STORE_CLASS_KEY", "io.delta.standalone.internal.storage.S3SingleDriverLogStore")
    DeltaSink.forRowData(new Path("s3a://myBucket/myTable"),
      hadoopConf,
      schema)
      .withPartitionColumns("date")
      .build()
  }
```

A redacted full stack trace is like the following
```
java.io.IOException: Cannot create file:/myTable/_delta_log at io.delta.standalone.internal.DeltaLogImpl.ensureLogDirectoryExist(DeltaLogImpl.scala:165) ~[?:?]at io.delta.standalone.internal.OptimisticTransactionImpl.prepareCommit(OptimisticTransactionImpl.scala:259) ~[?:?]at io.delta.standalone.internal.OptimisticTransactionImpl.commit(OptimisticTransactionImpl.scala:113) ~[?:?]at io.delta.flink.sink.internal.committer.DeltaGlobalCommitter.doCommit(DeltaGlobalCommitter.java:318) ~[?:?]at io.delta.flink.sink.internal.committer.DeltaGlobalCommitter.commit(DeltaGlobalCommitter.java:221) ~[?:?]at org.apache.flink.streaming.runtime.operators.sink.StreamingGlobalCommitterOperator.commit(StreamingGlobalCommitterOperator.java:88) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.operators.sink.AbstractStreamingCommitterOperator.commitUpTo(AbstractStreamingCommitterOperator.java:155) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.operators.sink.AbstractStreamingCommitterOperator.notifyCheckpointComplete(AbstractStreamingCommitterOperator.java:137) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.operators.sink.StreamingGlobalCommitterOperator.notifyCheckpointComplete(StreamingGlobalCommitterOperator.java:104) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.notifyCheckpointComplete(StreamOperatorWrapper.java:99) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.SubtaskCheckpointCoordinatorImpl.notifyCheckpointComplete(SubtaskCheckpointCoordinatorImpl.java:334) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.notifyCheckpointComplete(StreamTask.java:1178) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointCompleteAsync$10(StreamTask.java:1143) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.lambda$notifyCheckpointOperation$12(StreamTask.java:1166) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor$SynchronizedStreamTaskActionExecutor.runThrowing(StreamTaskActionExecutor.java:93) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.mailbox.Mail.run(Mail.java:90) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMailsWhenDefaultActionUnavailable(MailboxProcessor.java:344) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.processMail(MailboxProcessor.java:330) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:202) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:681) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.executeInvoke(StreamTask.java:636) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.runWithCleanUpOnFail(StreamTask.java:647) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:620) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:784) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at org.apache.flink.runtime.taskmanager.Task.run(Task.java:571) ~[flink-dist_2.12-1.13.2.jar:1.13.2]at java.lang.Thread.run(Thread.java:829) ~[?:?]
--
```
After some tracing, I believe that the problem may be in the [DeltaGlobalCommiter](https://github.com/delta-io/connectors/blob/925bd79bcae9b0c85e7cd7342e4686e7be2a21d5/flink/src/main/java/io/delta/flink/sink/internal/committer/DeltaGlobalCommitter.java#L215). I tested by changing the initialization of the DeltaLog to  `DeltaLog.forTable(conf, basePath.toString());` and seems to be working.

Let me know if you need any other information that could help resolving this issue.

Thanks in advance!

**Comment by allisonport-db** [comment]
Your fix makes sense to me, it seems [uri.getPath()](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/core/fs/Path.java#L304) only returns the path component of the uri.

**Comment by tdas** [comment]
@pabloFloresHdz would be interested in contributing this fix?

**Comment by pabloFloresHdz** [comment]
Thanks for your input @allisonport-db! @tdas I was waiting for more input from the team in case a more robust solution was necessary as I'm quite new to the ecosystem, but I can definitely open a PR to solve this

**Comment by allisonport-db** [comment]
Hi @pabloFloresHdz, when you have the time could you try running your code (without the fix in #341) but set `fs.defaultFS` set to `s3a://<bucket>`?

**Comment by pabloFloresHdz** [comment]
> Hi @pabloFloresHdz, when you have the time could you try running your code (without the fix in #341) but set `fs.defaultFS` set to `s3a://<bucket>`?

Using your suggestion I can see the files being written to the _delta_log s3 location without issues

**Comment by allisonport-db** [comment]
Cool thanks @pabloFloresHdz.

For anyone else encountering this issue, we will be doing a patch release with the fix soon. But the current workaround is setting `fs.defaultFS` to your intended file system.
